### PR TITLE
Game functionality created to be playable

### PR DESCRIPTION
**Pull request by JohannRodriguez**
# Game Logic

- The current player turn is displayed.
- You can put a number and it will change on the board.
- Winer's name displayed.
- If it's a draw a message is displayed.

### CODE REVIEW

File: lib/game.rb [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+    end
+  end
+
+  def array_include?(arg)
+    arg.length.times do |n|
+      return false unless include?(arg[n])
+    end
+    true
+  end
+end
```
Comment (whiz25) [inline lib/game.rb:21]: - [ ] put this module in it's on file. you need to separate concerns

File: lib/game.rb [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+
+# print boars for the user to see
+class UserDisplay
+  def self.print_board(grid)
+    p '********************'
+    grid.length.times { |n| p grid[n] }
+    p '********************'
```
Comment (whiz25) [inline lib/game.rb:28]: - [ ] this print code should live in the `main.rb` file

File: lib/game.rb [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+  end
+
+  def wich_turn(player_1_name, player_2_name)
+    player_turn_name = player_1_name if $current_turn.even?
+    mark = 'X' if $current_turn.even?
+    player_turn_name = player_2_name if $current_turn.odd?
+    mark = 'O' if $current_turn.odd?
+    p "It's #{player_turn_name}'s turn' ('#{mark}'), make your move:"
+  end
```
Comment (whiz25) [inline lib/game.rb:37]: - [ ] put this code in the `main.rb` file. this related to user interface logic

File: lib/game.rb [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+      sub_array.length.times do |n|
+        if player_mark_print == sub_array[n]
+          sub_array[n] = player_turn_print
+          $current_turn += 1
+          return modified_grid_print
+        end
+      end
+    end
+  end
+end
```
Comment (whiz25) [inline lib/game.rb:76]: - [ ] player logic should live in a separate file from the game logic. This file should only contain one class. The game class

File: lib/game.rb [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+  @@player_2_moves = []
+  def player_move_save(player_input, player_1_name, player_2_name)
+    @@player_1_moves.push(player_input) if $current_turn.even?
+    @@player_2_moves.push(player_input) if $current_turn.odd?
+    win(player_1_name, player_2_name)
+  end
+
+  def draw(no_moves)
+    $game_on = false if no_moves.length.zero?
+    return p "It's a draw, no winners" if no_moves.length.zero?
```
Comment (kobiyoyo) [inline lib/game.rb:14]: No `puts/print` and `get.chomp` outside `main.rb`

File: lib/game.rb [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+  end
+
+  def win(player_1, player_2)
+    8.times do |n|
+      if @@player_1_moves.array_include?(@@winning_cases[n])
+        p "#{player_1} wins!"
```
Comment (kobiyoyo) [inline lib/game.rb:20]: No `puts/print` and `get.chomp` outside `main.rb`

File: lib/game.rb [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+        $game_on = false
+        $game_on
+      end
+      if @@player_2_moves.array_include?(@@winning_cases[n])
+        p "#{player_2} wins!"
```
Comment (kobiyoyo) [inline lib/game.rb:25]: No `puts/print` and `get.chomp` outside `main.rb`

**Pull request merged**